crypto/signing/ed25519/suite.go

```diff
@@ -0,0 +1,91 @@
+package ed25519
+
+import (
+	"crypto/cipher"
+	"crypto/ed25519"
+
+	"github.com/ElrondNetwork/elrond-go/crypto"
+)
+
+// ED25519 is the string representations of the ed25519 scheme
+const ED25519 = "Ed25519"
+
+type suiteEd25519 struct {}
+
+// NewEd25519 returns a wrapper over ed25519
+func NewEd25519() *suiteEd25519 {
+	return &suiteEd25519{}
+}
+
+// CreateKeyPair returns a pair of Ed25519 keys
+func (s *suiteEd25519) CreateKeyPair(stream cipher.Stream) (crypto.Scalar, crypto.Point) {
+	publicKey, privateKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		panic("could not create ed25519 key pair: " + err.Error())
+	}
+
+	return &ed25519Scalar{privateKey}, &ed25519Point{publicKey}
+}
+
+// CreatePoint returns a newly created public key which is a point on ed25519
+func (s *suiteEd25519) CreatePoint() crypto.Point {
+	_, publicKey := s.CreateKeyPair(nil)
+	return publicKey
+}
+
+// GetPublicKeyPoint returns a Point that is the representation of a public key corresponding
+//  to the provided Scalar in the ed25519 signature scheme
+func (s *suiteEd25519) GetPublicKeyPoint(scalar crypto.Scalar) (crypto.Point, error) {
+	privateKey, ok := scalar.GetUnderlyingObj().(ed25519.PrivateKey)
+	if !ok {
+		return nil, crypto.ErrInvalidPrivateKey
+	}
+
+	publicKey, ok := privateKey.Public().(ed25519.PublicKey)
+	if !ok {
+		return nil, crypto.ErrGeneratingPubFromPriv
+	}
+
+	return &ed25519Point{publicKey}, nil
+}
+
+// String returns the string for the group
+func (s *suiteEd25519) String() string {
+	return ED25519
+}
+
+// ScalarLen returns the length of the ed25519 private key - which is the number of bytes of
+//  the seed (the actual private key) + the number of bytes of the public key
+func (s *suiteEd25519) ScalarLen() int {
+	return ed25519.PrivateKeySize
+}
+
+// CreateScalar creates a new Scalar which represents the ed25519 private key
+func (s *suiteEd25519) CreateScalar() crypto.Scalar {
+	_, privateKey, err := ed25519.GenerateKey(nil)
+	if err != nil {
+		panic("could not create ed25519 private key: " + err.Error())
+	}
+
+	return &ed25519Scalar{privateKey}
+}
+
+// PointLen returns the number of bytes of the ed25519 public key
+func (s *suiteEd25519) PointLen() int {
+	return ed25519.PublicKeySize
+}
+
+// GetUnderlyingSuite returns nothing because this is not a wrapper over another suite implementation
+func (s *suiteEd25519) GetUnderlyingSuite() interface{} {
+	return nil
```

Comment:
maybe log that this func should not be called.

Reply:
Done